src/Symfony/Component/DependencyInjection/ParameterBag/ParameterResolver.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\ParameterBag;
+
+use Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException;
+
+/**
+ * {@inheritdoc}
+ * This class doesn't support nested resolution (array('foo' => '%bar%', 'bar' => '%foobar%')). This implies that the getter returns a resolved value.
+ *
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+final class ParameterResolver implements ParameterResolverInterface
```

Comment:
This class is here to be defined as a service in third party bundles.
IMO we should avoid passing the container to services as much as possible so that way, we only inject it once.

In fact, we can even add this service in the ``FrameworkBundle``, what do you think?